endpoints/resources/src/main.cpp

```diff
@@ -516,6 +604,14 @@ namespace
 				else if (const auto resc = adm::client::resource_info(conn, name_iter->second); resc) {
 					exists = true;
 
+					std::string_view status = "?";
```

Comment:
The other member of the `resource_status` enumeration is `unknown`. Would it make more sense to default to "unknown" instead of "?"?

Reply:
question mark makes me... twitchy as well.

Reply:
iRODS allows admins to set the status to any string AFAIK. From the help text of `iadmin modresc`.
```
Setting the resource status to 'down' will cause iRODS to ignore that
resource and bypass communications with that server.  'up' or other strings
without 'down' in them will restore use of the resource.  'auto' will allow
the Resource Monitoring System (if running) to set the resource status
to 'auto-up' or 'auto-down'.
```
Based on that text, the question mark should be replaced with `up`. The resource administration lib may need a tweak to better handle the status too. Perhaps it shouldn't interpret the status string. I'll review that implementation and see what it does exactly.

I'll need to think through whether `unknown` works.

Reply:
Here's another thing to consider - should modifying the status be exposed at all?

Reply:
That's a good question... And if it is exposed, do we support "any string that doesn't contain 'down'" as "up"? Do we support "auto"?

Reply:
Yes, the question mark is not a problem, just feels weird.   'unknown' captures that sense better than a symbol, in my opinion.

> should modifying the status be exposed at all? 

Administrators could be using that field for any number of things.   We should not remove it.

Reply:
If we use "unknown", that means the resource could be up or down. That's my only hesitation about using that.

With that said, seeing "unknown" could make the admin reconsider the values they pass for the status when using tools other than the HTTP API.

Reply:
After some discussion, we've decided to return the resource status as is.

The HTTP API will continue to constrain the acceptable values for resource status to up and down.

Reply:
Done.